tutorials/10-bayesian-differential-equations/10_bayesian-differential-equations.jmd

```diff
@@ -13,312 +14,360 @@ using DifferentialEquations
+We generate noisy observations to use for the parameter estimation tasks in this tutorial.
+With the [`saveat` argument](https://docs.sciml.ai/latest/basics/common_solver_opts/) we specify that the solution is stored only at `0.1` time units.
+To make the example more realistic we add random normally distributed noise to the simulation.
 
 ```julia
-sol1 = solve(prob1, Tsit5(); saveat=0.1)
-odedata = Array(sol1) + 0.8 * randn(size(Array(sol1)))
-plot(sol1; alpha=0.3, legend=false);
-scatter!(sol1.t, odedata');
+sol = solve(prob, Tsit5(); saveat=0.1)
+odedata = Array(sol) + 0.8 * randn(size(Array(sol)))
+
+# Plot simulation and noisy observations.
+plot(sol; alpha=0.3)
+scatter!(sol.t, odedata'; color=[1 2], label="")
 ```
 
 ## Direct Handling of Bayesian Estimation with Turing
 
 Previously, functions in Turing and DifferentialEquations were not inter-composable, so Bayesian inference of differential equations needed to be handled by another package called [DiffEqBayes.jl](https://github.com/SciML/DiffEqBayes.jl) (note that DiffEqBayes works also with CmdStan.jl, Turing.jl, DynamicHMC.jl and ApproxBayes.jl - see the [DiffEqBayes docs](https://docs.sciml.ai/latest/analysis/parameter_estimation/#Bayesian-Methods-1) for more info).
 
-From now on however, Turing and DifferentialEquations are completely composable and we can write of the differential equation inside a Turing `@model` and it will just work. Therefore, we can rewrite the Lotka Volterra parameter estimation problem with a Turing `@model` interface as below:
+Nowadays, however, Turing and DifferentialEquations are completely composable and we can just simulate differential equations inside a Turing `@model`.
+Therefore, we write the Lotka-Volterra parameter estimation problem using the Turing `@model` macro as below:
 
 ```julia
-Turing.setadbackend(:forwarddiff)
-
-@model function fitlv(data, prob1)
-    σ ~ InverseGamma(2, 3) # ~ is the tilde character
+@model function fitlv(data, prob)
+    # Prior distributions.
+    σ ~ InverseGamma(2, 3)
     α ~ truncated(Normal(1.5, 0.5), 0.5, 2.5)
     β ~ truncated(Normal(1.2, 0.5), 0, 2)
     γ ~ truncated(Normal(3.0, 0.5), 1, 4)
     δ ~ truncated(Normal(1.0, 0.5), 0, 2)
 
+    # Simulate Lotka-Volterra model. 
     p = [α, β, γ, δ]
-    prob = remake(prob1; p=p)
-    predicted = solve(prob, Tsit5(); saveat=0.1)
+    predicted = solve(prob, Tsit5(); p=p, saveat=0.1)
 
+    # Observations.
     for i in 1:length(predicted)
-        data[:, i] ~ MvNormal(predicted[i], σ)
+        data[:, i] ~ MvNormal(predicted[i], σ^2 * I)
     end
+
+    return nothing
 end
 
-model = fitlv(odedata, prob1)
+model = fitlv(odedata, prob)
 
-# This next command runs 3 independent chains without using multithreading.
+# Sample 3 independent chains with forward-mode automatic differentiation (the default).
```

Comment:
Maybe add a link to the docs on how to change AD backends here?

Reply:
It's shown how to change it (and change it back) below, is that sufficiently clear? Or should we add a link here? Or maybe rather below?

Reply:
thanks for the clarification, that should be sufficient.